lib/iiif_print/split_pdfs/base_splitter.rb

```diff
@@ -5,10 +5,14 @@
 
 module IiifPrint
   module SplitPdfs
-    # The purpose of this class is to split the PDF into constituent TIFF files.
+    # @abstract
+    #
+    # The purpose of this class is to split the PDF into constituent image files.
     #
     # @see #each
-    class PagesIntoImagesService
+    class BaseSplitter
+      class_attribute :image_extension
+
       DEFAULT_COMPRESSION = 'lzw'.freeze
```

Comment:
I'm wondering if this should be moved to the `PagesToTiffsSplitter`, LZW is a tiff compression and won't apply to PNGs

Reply:
What would be the compression that we use for PNG?

Reply:
We use it in the general case [here](https://github.com/scientist-softserv/iiif_print/blob/711583652618e7ec7df8714114f4f23ca5ccdda1/lib/iiif_print/split_pdfs/base_splitter.rb#L63-L65):

```ruby
        cmd = "gs -dNOPAUSE -dBATCH -sDEVICE=#{gsdevice} " \
              "-dTextAlphaBits=4 -sCompression=#{compression} " \
              "-sOutputFile=#{output_base} -r#{ppi} -f #{pdfpath}"
```

My strong suspicion is that we should extract a `compression` method and also provide `compression?` and conditionally add that switch if we're going to perform compression.

Reply:
i agree with extracting compression out, i don't believe PNGs have a compression like tiffs

Reply:
it was originally not in the newspaper works gem code and we added it because a RAW tiff was just way too large